### DISCUSSION TITLE
Distributed deployment example

### DISCUSSION THREAD

**Discussion by victusfate** [online-ml/river#803] [Show and tell]
I wanted to share something I built on top of river for a continuous learning system. 

It leverages item ratings in realtime through:
- redis pubsub
- updates a recent ratings ordered set
- does a full training hourly via an exported csv
- delta trains from the recent ratings after the models last event

I don't see any major issues unless the delta training set grows large vs the memory cost (many millions fit into a reasonable set of memory).

https://github.com/victusfate/concierge

**Comment by MaxHalford** [comment]
_I love the smell of real-time in the morning._

That looks really fun, I love it! We definitely need more user examples like these. Thank you very much for sharing, I hope it will inspire others :)
